src/components/RealmSelect/RealmSelect.tsx

```diff
@@ -0,0 +1,137 @@
+import CloseIcon from '@mui/icons-material/Close'
+import { TabContext, TabList, TabPanel } from '@mui/lab'
+import {
+  AppBar,
+  Button,
+  Dialog,
+  IconButton,
+  Slide,
+  Tab,
+  Toolbar,
+  Typography
+} from '@mui/material'
+import { TransitionProps } from '@mui/material/transitions'
+import { first, xor } from 'lodash'
+import {
+  forwardRef,
+  ReactElement,
+  Ref,
+  SyntheticEvent,
+  useEffect,
+  useState
+} from 'react'
+import { useQuery } from 'react-query'
+
+import { getRealmSelectable } from '../../lib/queries/getRealmSelectable'
+import { GetRealmSelectableParams } from '../../lib/queries/getRealmSelectable/getRealmSelectable'
+
+import { RealmSelectSelect } from './Select'
```

Comment:
no. otherwise the external consumers would need to use RealmSelectDialog which doesn't make much sense because they think they're getting a select box.